### PR TITLE
feat: Optimise WebGL compositor

### PR DESCRIPTION
**Pull request by eXponenta**
Follow: #43 

Remove lookup texture dependency 
Use lazy uniform syncing
Reduce unnecessary texture activations
Minor CSS Fix

Benefits:
- max 3 (regularly is 2) versus 5 per frame GL operations
- faster color calculation  (not require extra texture sampling)
- uniform sync is faster than texture uploading

Before:
image

After:
image

### CODE REVIEW

File: runtimes/web/src/compositor.js [hunk 1]
```diff
@@ -108,27 +118,38 @@ export class WebGLCompositor {
...
+            table = this.table,
+            lastPallete = this.lastPallete,
+            rgb = this.palleteBuffer;
 
-        // Upload palette
-        // TODO(2021-08-08): Don't upload if the palette is unchanged
-        const rgb = new Uint8Array(colorBuffer.buffer, 3*4);
-        for (let ii = 0, n = 0; ii < 4; ++ii) {
+        // Upload palette when needed 
+        let syncPallete = false;
```
Comment (MaxGraey) [inline runtimes/web/src/compositor.js:129]:
```suggestion
        let syncPalette = false;
```

File: runtimes/web/src/compositor.js [hunk 2]
```diff
@@ -108,27 +118,38 @@ export class WebGLCompositor {
...
+
+        for (let ii = 0, n = 0; ii < PALLETE_SIZE; ++ii) {
```
Comment (MaxGraey) [inline runtimes/web/src/compositor.js:131]: PALETTE_SIZE

**Comment by aduros** [comment]
This looks incredible, nice work!

File: runtimes/web/src/styles.css [hunk 3]
```diff
@@ -1,3 +1,7 @@
+* {
+  box-sizing: border-box;
+}
+
```
[the remark this answers is not in the document]
Comment (eXponenta) [reply runtimes/web/src/styles.css:4]: Yep, because your player has a padding, without it, player will be have extra gap from right side.

**Comment by aduros** [comment]
Thanks!

**Pull request merged**